File: src/it/echo-mojo-java/src/main/java/org/scala_tools/mojo/EchoMojo.java

package org.scala_tools.mojo;

/*
 * Echo mojo in java for testing the passng of properties from the command line
 */

import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugin.MojoExecutionException;
import org.apache.maven.plugin.MojoFailureException;

/**
 * Echos an object string to the output screen.
 * @goal echo
 * @requiresProject false
 */
public class EchoMojo extends AbstractMojo
{
    /**
     * Any Object to print out.
     * @parameter expression="${echo.message}" default-value="Hello World..."
     */
    private Object message;

    public void execute()
        throws MojoExecutionException, MojoFailureException
    {
        getLog().info( message.toString() );
    }
}

File: src/it/echo-mojo-scala/src/main/scala/org/scala_tools/mojo/EchoMojo.scala
package org.scala_tools.mojo

/*
 * Echo mojo in java for testing the passng of properties from the command line
 */

import org.apache.maven.plugin._
import org.scala_tools.maven.mojo.annotations._
import org.apache.maven.project.MavenProject

@goal("echo")
@requiresProject(false)
class EchoMojo extends AbstractMojo
{
    @parameter
    @expression("${echo.message}")
    // Not yet implemented @default("Hello World...")
    var message: String = "Hello World..."

    @throws(classOf[MojoExecutionException])
    override def execute() 
    {
        getLog().info( message.toString )
    }
}